### PR TITLE
fix(i18n): correctly compute the current locale

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes https://github.com/withastro/astro/issues/12185

## Testing

I added a new test case

## Docs

N/A

### CODE REVIEW

File: packages/astro/test/fixtures/i18n-routing-dynamic/src/pages/[language].astro [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+export async function getStaticPaths() {
+	return [{ params: { language: "ru" } }, { params: { language: "en" } }];
+}
+
+const { currentLocale } = Astro;
+console.log(currentLocale);
```
Comment (bluwy) [inline packages/astro/test/fixtures/i18n-routing-dynamic/src/pages/[language].astro:7]:
```suggestion
```

should we remove this?

File: pnpm-lock.yaml
Comment (bluwy) [inline pnpm-lock.yaml:1]: Ah I forgot to update the lockfile. Weird that there's not CI fail

File: packages/astro/src/core/render-context.ts [hunk 2]
```diff
@@ -537,10 +537,10 @@ export class RenderContext {
...
 		// and url.pathname to pass Astro.currentLocale tests.
 		// A single call with `routeData.pathname ?? routeData.route` as the pathname still fails.
-		return (this.#currentLocale ??=
-			computeCurrentLocale(routeData.route, locales, defaultLocale) ??
-			computeCurrentLocale(url.pathname, locales, defaultLocale) ??
-			fallbackTo);
+
+		return (this.#currentLocale ??= routeData.pathname
+			? computeCurrentLocale(routeData.pathname, locales, defaultLocale)
+			: (computeCurrentLocale(url.pathname, locales, defaultLocale) ?? fallbackTo));
```
Comment (bluwy) [inline packages/astro/src/core/render-context.ts:543]: Should the `fallbackTo` apply to both the two results of the ternary? Right now it's only applying to if `route.pathname` is falsy.

Comment (matthewp) [reply packages/astro/src/core/render-context.ts:543]: I think this is getting difficult to read. Can we convert this to boring `if/else` statements with assignment?

**Pull request merged**